### DISCUSSION TITLE
[python] SqlInjectionSink missing results

### DISCUSSION THREAD

**Discussion by ChewZL** [github/securitylab#122] [General] [answered]
Hi, I was exploring QL for python and noticed that SqlInjectionSink, defined in the [Sql.qll](https://github.com/github/codeql/blob/master/python/ql/src/semmle/python/security/injection/Sql.qll) library misses some results. Running the following query:
```
from TaintSink sink
where sink instanceof SqlInjectionSink
select sink
```
misses out certain results like cursor.execute() at [/django/db/models/sql/compiler.py line 1161](https://lgtm.com/projects/g/django/django/snapshot/81d3e57b07690f316543e72740c2d6c09e8ebb7f/files/django/db/models/sql/compiler.py?sort=name&dir=ASC&mode=heatmap#L1156)

Does anyone know the reason for this, and is there any way to include it as a sink?
Thanks!

**Comment by RasmusWL** [comment, marked as answer]
Hi @ChewZL, thanks for reporting this missing result :+1:

To handle this requires better QL modeling of libraries. Currently we're doing a major rewrite of the Python analysis, and won't have time to look into this in the next couple of months. I'll note it down so we remember to look into it afterwards :+1:
